### PR TITLE
Fix #4375: Always use trainer.global_step for step

### LINKED ISSUE #4375: LearningRateMonitor produces inconsistent logs with logging_interval="epoch"
## 🐛 Bug

`LearningRateMonitor` uses `step=trainer.current_epoch` whereas in other places, it is always `step=trainer.global_step`.

This creates inconsistencies and makes the log hard to process:

https://colab.research.google.com/drive/1bucI1oGCc_xNsnP_lvBWTz3x_bauH3o7

```
[{'lr-SGD': 0.1, 'step': 0},
 {'epoch': 0, 'step': 49, 'training_loss': 6.862762529635802e-05},
 {'epoch': 0, 'step': 99, 'training_loss': 4.8670010244222794e-09},
 {'epoch': 0, 'step': 149, 'training_loss': 1.5898393712632242e-13},
 {'epoch': 0, 'step': 199, 'training_loss': 4.135580766728708e-14},
 {'epoch': 0, 'step': 249, 'training_loss': 5.2458037913538647e-14},
 {'epoch': 0, 'step': 299, 'training_loss': 4.879430193227563e-14},
 {'epoch': 0, 'step': 312, 'validation_loss': 4.705802669193945e-14},
 {'lr-SGD': 0.010000000000000002, 'step': 1},
 {'epoch': 1, 'step': 349, 'training_loss': 4.884981308350689e-14},
 {'epoch': 1, 'step': 399, 'training_loss': 4.6407322429331543e-14},
 {'epoch': 1, 'step': 449, 'training_loss': 4.679590048795035e-14},
 {'epoch': 1, 'step': 499, 'training_loss': 4.3798298321462426e-14},
 {'epoch': 1, 'step': 549, 'training_loss': 4.368727601899991e-14},
 {'epoch': 1, 'step': 599, 'training_loss': 4.335420911161236e-14},
 {'epoch': 1, 'step': 625, 'validation_loss': 4.4799982544226416e-14},
 {'lr-SGD': 0.0010000000000000002, 'step': 2},
 {'epoch': 2, 'step': 649, 'training_loss': 4.5352610555937645e-14},
 {'epoch': 2, 'step': 699, 'training_loss': 4.551914400963142e-14},
 {'epoch': 2, 'step': 749, 'training_loss': 4.4853010194856324e-14},
 {'epoch': 2, 'step': 799, 'training_loss': 4.2021941482062175e-14},
 {'epoch': 2, 'step': 849, 'training_loss': 4.651834473179406e-14},
 {'epoch': 2, 'step': 899, 'training_loss': 4.296563105299356e-14},
 {'epoch': 2, 'step': 938, 'validation_loss': 4.4716627725953015e-14},
 {'fake_test_acc': 4.718447854656915e-14, 'step': 939}]
```

For `lr-SGD` rows, `step` is the epoch, otherwise, `step` is the global step.

### Expected behavior

I expect `step` to be used consistently, i.e. it should be always be `step=trainer.global_step`.
Additionally, `LearningRateMonitor` could provide a value for `epoch=trainer.current_epoch`.

### Environment

See Colab notebook.

### PR DESCRIPTION
**Pull request by moi90**
## What does this PR do?

This PR changes LearningRateMonitor to consistently use `step=trainer.global_step`, independently of `logging_interval`.

Fixes #4375

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.    
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

**Comment by rohitgr7** [comment]
Is there a problem logging it with epoch in case of epoch logging?? The reason I am asking this because it might not align with the global_step of other logs which are logged at epoch level since those are created at epoch end and this one at epoch start and the logged global_step might be +1 as compared to the other ones.

**Comment by moi90** [comment]
I think it is right to log the learning rate at the beginning of the epoch with the `trainer.global_step` at this time. It is only natural to read the log in chronological order and assume that logged properties like learning rate apply to all following steps until a new value is logged. Anything else does not make sense at all in my understanding.

**Comment by edenlightning** [comment]
@moi90 would you be able to take a look at failing tests?

**Comment by moi90** [comment]
It seems that the failing tests are fixed by the changes merged from master.

File: pytorch_lightning/callbacks/lr_monitor.py [hunk 1]
```diff
@@ -106,6 +106,7 @@ def on_train_batch_start(self, trainer, *args, **kwargs):
             latest_stat = self._extract_stats(trainer, interval)
 
             if trainer.logger is not None and latest_stat:
+                latest_stat.setdefault("epoch", trainer.current_epoch)
```
Comment (SkafteNicki) [inline pytorch_lightning/callbacks/lr_monitor.py:109]: Is it necessary to add the epoch since we already as default logs it?

Comment (moi90) [reply pytorch_lightning/callbacks/lr_monitor.py:109]: Yes, you're right. I will remove that.

**Review by rohitgr7 — APPROVED** [review]
Tested. LGTM!

**Review by awaelchli — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]
LGTM

**Comment by moi90** [comment]
Great, thanks for merging!
